src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -802,5 +806,10 @@ internal static IGrainExtensionMethodInvoker TryGetExtensionInvoker(GrainTypeMan
                 $"Provider extension handler type {handlerType} was not found in the type manager",
                 nameof(handlerType));
         }
+
+        public void Participate(ISiloLifecycle lifecycle)
+        {
+            lifecycle.Subscribe(SiloLifecycleStage.RuntimeInitialize, OnRuntimeInitializeStart, OnRuntimeInitializeStop);
```

Comment:
Using the same stage would mean that it will run in parallel with the InsideRuntimeClient, correct? Couldn't that cause issues?

Reply:
This means that the Inside runtime client receives its OnRuntimeInitializeStart and OnRuntimeInitializeStop calls in  parallel with everything else registered at the RuntimeInitialize stage, which, at this time, is only the silo.